pipelines/batch/src/main/java/org/openmrs/analytics/ResourceTag.java

```diff
@@ -32,4 +32,20 @@ public class ResourceTag implements Serializable {
   Coding coding;
   String resourceId;
   Integer tagType;
+
+  @Override
+  public boolean equals(Object other) {
```

Comment:
I think it is better to switch to `AutoValue` as it takes care of details and boilerplate for pure data objects. For example, I think you need to override `hashCode` as well when you change `equals`. Here is one example for [`SearchSegmentDescriptor`](https://github.com/google/fhir-data-pipes/blob/1f2dff7d9845e7cef21b333c63630e58f32fd78b/pipelines/batch/src/main/java/org/openmrs/analytics/SearchSegmentDescriptor.java#L25).

Reply:
So, I tried AutoValue and it generates this code within AutoValue_ResourceTag:

`@Override
  public boolean equals(Object o) {
    if (o == this) {
      return true;
    }
    if (o instanceof ResourceTag) {
      ResourceTag that = (ResourceTag) o;
      return this.coding.equals(that.coding())
          && this.resourceId.equals(that.resourceId())
          && this.tagType.equals(that.tagType());
    }
    return false;
  }`

And the pipeline still throws warning:

`22:08:41.688 [direct-runner-worker] WARN  o.a.beam.sdk.util.MutationDetectors o.a.b.s.u.MutationDetectors$CodedValueMutationDetector.verifyUnmodifiedThrowingCheckedExceptions:145 - Coder of type class org.apache.beam.sdk.coders.KvCoder has a #structuralValue method which does not return true when the encoding of the elements is equal. Element KV{18534, ResourceTag{coding=org.hl7.fhir.r4.model.Coding@1ae2bb4a, resourceId=18534, tagType=1}}`

Reply:
Added hashcode method

Reply:
yeah